syntax = "proto2";

package chat;

message Response {
  required string msg = 1;
  required int64 userId = 2;
  required bool success = 3;
}

message Register {
  required string email = 1;
  required string plaintextPassword = 2;
}

message Authenticate {
  required string email = 1;
  required string plaintextPassword = 2;
}

message JoinRoom {
  required int64 userId = 1;
  required int64 roomId = 2;
}

message LeaveRoom {
  required int64 userId = 1;
  required int64 roomId = 2;
}

message ChatMessage {
  required int64 userId = 1;
  required int64 roomId = 2;
  required string msg = 3;
}
